web-api/src/ingestion/routes.rs

```diff
@@ -134,50 +152,77 @@ async fn new_documents(
 
     let start = Instant::now();
 
-    let (documents, mut failed_documents) = body
+    let (preprocessed, mut failed_documents): (Vec<_>, Vec<_>) = body
```

Comment:
:art: it's nicer to annotate the types at the end of the call like `.partition_result::<Vec<_>, Vec<_>, _, _>()` such that the IDE can show the full type hint for the assigned variables.

Reply:
I would argue it's the other way around as it's also clearer what parts the type hints are exactly for.

And at least in my case the IDE is showing type hints properly :thinking:

Reply:
exactly, sry i should have said that i mean the inlay hints. tool tips always show properly.